packages/lockfile-lint-api/src/ParseLockfile.js

```diff
@@ -124,26 +125,26 @@ class ParseLockfile {
     }
   }
 
-  _flattenNpmDepsTree (npmDepsTree) {
-    let flattenedDepTree = {}
-    let flattenedNestedDepsTree = {}
+  _flattenNpmDepsTree (npmDepsTree, npmDepMap = {}) {
     for (const [depName, depMetadata] of Object.entries(npmDepsTree)) {
       const depMetadataShortend = {
         version: depMetadata.version,
-        resolved: depMetadata.resolved ? depMetadata.resolved : depMetadata.version,
+        resolved: depMetadata.resolved,
```

Comment:
@JamesSingleton can you comment why was this changed?
IIRC this was supposed to solve cases where a resolved property may not be there but a version would, for cases like local file URLs or something like this.

Reply:
So this is due to the `new URL(metadata.resolved)` failing if this is a version?
I'll track back the source of this and we'll find out.

Reply:
From https://github.com/lirantal/lockfile-lint/issues/49#issuecomment-578963073

> The error points to this line
```let packageResolvedURL = {}
      try {
        packageResolvedURL = new URL(packageMetadata.resolved)
      } catch (error) {
        throw new PackageError(packageName, error)
      }
```
> Not sure how this is now just coming up now though. Looks like this will happen with any dependency that does not have a true URL for resolved since resolved is optional and doing new URL("1.1.1") is not a valid URL.

Sounds like you're saying this is a general bug unrelated to this change.
I tracked the source of using the version to a `package-lock.json` structure where the `version` field is used to specify the git URL, see here: https://github.com/lirantal/lockfile-lint/commit/a48311598b7f7e3e091d56e26d801894c00eb02b#diff-24ebf59c7b0eeb6040c5311942ef4a3eR16

Reply:
@JamesSingleton nevermind here since we already have this discussion going on #54